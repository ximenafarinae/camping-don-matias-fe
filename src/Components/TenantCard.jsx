import React from "react";
import { Card, Button, ListGroup, ListGroupItem } from "react-bootstrap";
import moment from "moment";

const TenantCard = (props) => {
  const id = props.id;
  const {
    name,
    last_name,
    birth_date,
    dni,
    vehicle_registration_number,
    phone,
    email,
    address,
  } = props.data;
  return (
    <Card>
      <Card.Header className="text-success font-weight-bold justify-content-center">
        INQUILINO
      </Card.Header>
      <Card.Body>
        <ListGroup className="list-group-flush">
          {" "}
          <ListGroupItem>
            <span className="text-success font-weight-bold">
              Nombre y Apellido:
            </span>
            <span>
              {" "}
              {name} {last_name}
            </span>
          </ListGroupItem>
          <ListGroupItem>
            <span className="text-success font-weight-bold">Fec. de Nac.:</span>
            {"  "}
            <span>
              {birth_date ? moment(birth_date).format("DD/MM/YYYY") : ""}
            </span>
          </ListGroupItem>
          <ListGroupItem>
            <span className="text-success font-weight-bold">DNI:</span>{" "}
            <span> {dni}</span>
          </ListGroupItem>
          <ListGroupItem>
            <span className="text-success font-weight-bold">
              Datos del vehiculo:
            </span>{" "}
            <span> {vehicle_registration_number}</span>
          </ListGroupItem>
          <ListGroupItem>
            <span className="text-success font-weight-bold">Teléfono:</span>{" "}
            <span>{phone}</span>
          </ListGroupItem>
          <ListGroupItem>
            <span className="text-success font-weight-bold">
              Correo electrónico:
            </span>{" "}
            <span>{email}</span>
          </ListGroupItem>
          <ListGroupItem>
            <span className="text-success font-weight-bold">Dirección:</span>{" "}
            <span>{address}</span>
          </ListGroupItem>
        </ListGroup>
      </Card.Body>
      <Card.Body className="d-inline-flex justify-content-end">
        <Button
          size="sm"
          variant="outline-success"
          className="font-weight-bold"
          onClick={() => {
            props.history.push(`/edit-tenant/${id}`);
          }}
        >
          EDITAR
        </Button>
      </Card.Body>
    </Card>
  );
};

export default TenantCard;
