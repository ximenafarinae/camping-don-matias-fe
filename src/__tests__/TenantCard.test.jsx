import { shallow } from "enzyme";
import TenantCard from "../Components/TenantCard";

describe("<TenantCard></TenantCard>", () => {
  const props = {
    id: "81",
    data: {
      id: 81,
      name: "Pablo",
      last_name: "Escobar",
      birth_date: "1960-02-10T00:00:00.000Z",
      dni: "78945623",
      vehicle_registration_number: "AB308FC",
      phone: "2246 44 4365",
      email: "ejmplo@ejemplo.com",
      address: "Calle ejemplo",
    },
    history: { push: jest.fn() },
  };

  test("Se debe cargar el componente", () => {
    const wrapper = shallow(<TenantCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Se debe cargar los input con los valores del response", async () => {
    const wrapper = shallow(<TenantCard {...props} />);
    const nameValue = wrapper.find("span").at(1).text().trim();
    expect(nameValue).toBe("Pablo Escobar");
  });

  test("Se debe llamar a history.push al hacer click al editar", () => {
    const wrapper = shallow(<TenantCard {...props} />);
    const btnEditar = wrapper.find("Button").simulate("click");
    expect(props.history.push).toHaveBeenCalledTimes(1);
    expect(props.history.push).toHaveBeenCalledWith('/edit-tenant/81');

    
  });
});
