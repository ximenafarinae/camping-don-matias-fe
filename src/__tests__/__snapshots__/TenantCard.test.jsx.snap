// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TenantCard></TenantCard> Se debe cargar el componente 1`] = `ShallowWrapper {}`;
